integrations/opennms-wsman-asset-provisioning-adapter/src/main/java/org/opennms/netmgt/provision/WsManAssetProvisioningAdapter.java

```diff
@@ -0,0 +1,463 @@
+
+package org.opennms.netmgt.provision;
+
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.MissingFormatArgumentException;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import org.opennms.core.spring.BeanUtils;
+
+import org.opennms.core.utils.InetAddressUtils;
+import org.opennms.core.utils.PropertiesUtils;
+
+import org.opennms.core.wsman.WSManClient;
+import org.opennms.core.wsman.WSManClientFactory;
+import org.opennms.core.wsman.WSManConstants;
+import org.opennms.core.wsman.WSManEndpoint;
+import org.opennms.core.wsman.cxf.CXFWSManClientFactory;
+import org.opennms.core.wsman.exceptions.WSManException;
+
+import org.opennms.netmgt.config.wsmanAsset.adapter.AssetField;
+import org.opennms.netmgt.config.wsmanAsset.adapter.WqlObj;
+import org.opennms.netmgt.config.wsman.WsmanAgentConfig;
+import org.opennms.netmgt.config.WsManAssetAdapterConfig;
+import org.opennms.netmgt.dao.WSManConfigDao;
+import org.opennms.netmgt.dao.api.NodeDao;
+import org.opennms.netmgt.events.api.EventConstants;
+import org.opennms.netmgt.events.api.EventForwarder;
+import org.opennms.netmgt.events.api.annotations.EventHandler;
+import org.opennms.netmgt.events.api.annotations.EventListener;
+import org.opennms.netmgt.model.OnmsAssetRecord;
+import org.opennms.netmgt.model.OnmsIpInterface;
+import org.opennms.netmgt.model.OnmsNode;
+import org.opennms.netmgt.model.events.EventBuilder;
+import org.opennms.netmgt.model.events.snmp.SyntaxToEvent;
+
+import org.opennms.netmgt.xml.event.Event;
+import org.opennms.netmgt.xml.event.Parm;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeanWrapper;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.PropertyAccessorFactory;
+import org.springframework.transaction.TransactionStatus;
+import org.springframework.transaction.support.TransactionCallback;
+import org.springframework.util.Assert;
+
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import com.google.common.collect.Lists;
+
+@EventListener(name="WsManAssetProvisioningAdapter")
+public class WsManAssetProvisioningAdapter extends SimplerQueuedProvisioningAdapter {
+    private static final Logger LOG = LoggerFactory.getLogger(WsManAssetProvisioningAdapter.class);
+
+        private NodeDao m_nodeDao;
+        private EventForwarder m_eventForwarder;
+        private WsManAssetAdapterConfig m_config;
+        private WSManClientFactory m_factory = new CXFWSManClientFactory();
+       // private WSManClientFactory m_factory;
+        private WSManConfigDao m_wsManConfigDao;
+
+        /**
+         * Constant <code>NAME="WsManAssetProvisioningAdapter"</code>
+         */
+        public static final String NAME = "WsManAssetProvisioningAdapter";
```

Comment:
This doesn't need to be public or static.
```suggestion
         private final String NAME = "WsManAssetProvisioningAdapter";
```